core/src/specialized_solver/fast_monodomain_solver/fast_monodomain_solver_compute.tpp

```diff
@@ -372,12 +387,14 @@ void FastMonodomainSolverBase<
     // x_i = d'_i - c'_i * x_{i+1}
 
     // helper buffers c', d'
-    static std::vector<double> cAlgebraic(nValues - 1);
-    static std::vector<double> dAlgebraic(nValues);
+    std::vector<double> cAlgebraic(nValues - 1);
+    std::vector<double> dAlgebraic(nValues);
```

Comment:
I assume that the original motivation for the `static` was to reuse the same memory, which could save significant time from the allocation.

An alternative method could be to declare the vector outside the loop, `reserve` a reasonable max number, and `resize` inside the loop.

The current approach apparently works as well, just keep in mind that performance might be impacted. Could have also been premature optimization, but I would expect that repeatedly allocating is an issue.